src/semantic-router/pkg/extproc/req_filter_reason_test.go

```diff
@@ -284,6 +290,70 @@ func TestReasoningModeComprehensive(t *testing.T) {
 	}
 }
 
+func TestChatTemplateKwargsPreservedWhenTogglingReasoning(t *testing.T) {
+	router := &OpenAIRouter{
+		Config: &config.RouterConfig{
+			IntelligentRouting: config.IntelligentRouting{
+				ReasoningConfig: config.ReasoningConfig{
+					ReasoningFamilies: map[string]config.ReasoningFamilyConfig{
+						"qwen3": {
+							Type:      "chat_template_kwargs",
+							Parameter: "enable_thinking",
+						},
+					},
+				},
+			},
+			BackendModels: config.BackendModels{
+				ModelConfig: map[string]config.ModelParams{
+					"qwen3-model": {
+						ReasoningFamily: "qwen3",
+					},
+				},
+			},
+		},
+	}
+
+	makeBody := func() []byte {
+		b, _ := json.Marshal(map[string]interface{}{
```

Comment:
Nit: error was overlooked; recommended to use `require.NoError` to prevent silent failures.